### ISSUE TITLE
Security Error on Blackberry 10.3.3

### ISSUE THREAD

**Issue by philippseith** [siacs/Conversations#2250]
#### General information

* **Version:** 1.15.4
* **Device:** Blackberry Z30, OS 10.3.3.2163
* **Android Version:** 4.3 (Kernel 3.2.41-QNX)
* **Server name:** jabber.de
* **Server software:** ?
* **Installed server modules:** ?
* **Conversations source:** Play Store

#### Steps to reproduce

1. Update Blackberry OS from 10.3.2 to 10.3.3
2. Go to Manage accounts in Conversations
3. Formerly working account is marked with "Sicherheitsfehler"

Another jabber.de account, new to this device, is also marked with "Sicherheitsfehler", when added.
A newly created account at conversations.im is working

#### Expected result

The account should be marked "online"

#### Debug output

Currently not available. I am struggling setting up the Blackberry Plugin for Android Studio.

**Comment by iNPUTmice** [comment]
> Formerly working account is marked with "Sicherheitsfehler"

This can happen if they either downgraded the login mechanism (unlikely), or if you are victim of a MIM attack (very unlikely), or if they changed something with their cipher suite/TLS stack that makes it incompatible with this rather old operating system. You might want to contact them and ask them about this.

**Comment by ERFier** [comment]
I can confirm the problem also with mailbox.org XMPP-Account.

**Comment by mimi89999** [comment]
@ERFier same OS?

**Comment by ERFier** [comment]
Yes, also BlackBerry 10.3.3. and also since the last update from 10.3.2.

**Comment by mimi89999** [comment]
@ERFier mailbox.org does only TLS1.2. Android 4.3 doesn't support TLS1.2.

Mailbox.org ciphersuite: https://xmpp.net/result.php?domain=mailbox.org&type=client

Android 4.3 reference ciphersuite: https://www.ssllabs.com/ssltest/viewClient.html?name=Android&version=4.3&key=61

**Comment by iNPUTmice** [comment]
> Yes, also BlackBerry 10.3.3. and also since the last update from 10.3.2.

Seems more like a BlackBerry issue then. Maybe they removed some ciphers.

**Comment by BlauerHunger** [comment]
The BB Android Runtime does only support TLS 1.0.

**Comment by ERFier** [comment]
ok so it seams to be just an unhappy coincidence of the BB update and them dropping support for TLS1.0.

Thanks for clearification.

**Comment by philippseith** [comment]
Coincidence with two different servers (jabber.de/mailbox.org) and two different BB update times (yours/mine)?  I am not convinced.

**Comment by philippseith** [comment]
I tried to find out myself which TLS versions are supported in the BB Android runtime.
ssllabs.com with Firefox 51

![img_20170301_175930](https://cloud.githubusercontent.com/assets/6890503/23472248/7ea32ba6-feac-11e6-9aa8-998260232bb2.png)

Seems to support TLS 1.2. But is it possible that Firefox brought the support by itself onto BB?

So I installed this: 
[https://play.google.com/store/apps/details?id=org.aseemsethi.sslTestTool](url)
It also tells that TLS 1.2 is working:

![img_20170301_181756](https://cloud.githubusercontent.com/assets/6890503/23472418/0598c86e-fead-11e6-9891-b3537775bf4d.png)

Does this prove the general TLS 1.2 supportin BBs android runtime? I am not sure. So it does not prove wrong that the missing support is the cause of the problem. But another fact makes me wonder. The server where I had problems with was jabber.de. And this server supports TLS 1.0:
[https://xmpp.net/result.php?domain=jabber.de&type=client#tls](url)
